packit/schema.py

```diff
@@ -1,82 +1,280 @@
+    """
+
+    def _serialize(self, value: typing.Any, attr: str, obj: typing.Any, **kwargs):
+        raise NotImplementedError
+
+    def _deserialize(
+        self,
+        value: typing.Any,
+        attr: typing.Optional[str],
+        data: typing.Optional[typing.Mapping[str, typing.Any]],
+        **kwargs,
+    ) -> SyncFilesItem:
+        if isinstance(value, dict):
+            try:
+                if not isinstance(value["src"], str):
+                    raise ValidationError("src have to be str")
+                if not isinstance(value["dest"], str):
+                    raise ValidationError("dest have to be str")
+                file_to_sync = SyncFilesItem(src=value["src"], dest=value["dest"])
+            except KeyError as e:
+                raise ValidationError(e.__repr__())
+
+        elif isinstance(value, str):
+            file_to_sync = SyncFilesItem(src=value, dest=value)
+
+        else:
+            raise ValidationError("Invalid data provided. str/dict required")
+
+        return file_to_sync
+
+
+class ActionField(fields.Field):
+    """
+    Field class representing Action.
+    """
+
+    def _serialize(self, value: typing.Any, attr: str, obj: typing.Any, **kwargs):
+        raise NotImplementedError
+
+    def _deserialize(
+        self,
+        value: typing.Any,
+        attr: typing.Optional[str],
+        data: typing.Optional[typing.Mapping[ActionName, typing.Any]],
+        **kwargs,
+    ) -> typing.Dict:
+        if not isinstance(value, dict):
+            raise ValidationError("Invalid data provided. dict required")
+
+        self.validate_all_actions(actions=list(value))
+        data = {}
+        for key, val in value.items():
+            data.update({ActionName(key): val})
+
+        return data
+
+    def validate_all_actions(self, actions: list) -> None:
+        """
+        Validates all keys and raises exception with list of all invalid keys
+        """
+        invalid_actions = []
+        for action in actions:
+            if not ActionName.is_valid_action(action):
+                invalid_actions.append(action)
+
+        if invalid_actions:
+            raise ValidationError(f"Unknown action(s) provided: {invalid_actions}")
+
+
+class NotProcessedField(fields.Field):
+    """
+    Field class to mark fields which wil not be processed, only generates warning.
+    Can be passed additional message via additional_message parameter.
+
+    :param str additional_message: additional warning message to be displayed
+    """
+
+    def _serialize(self, value: typing.Any, attr: str, obj: typing.Any, **kwargs):
+        raise NotImplementedError
+
+    def _deserialize(
+        self,
+        value: typing.Any,
+        attr: typing.Optional[str],
+        data: typing.Optional[typing.Mapping[str, typing.Any]],
+        **kwargs,
+    ):
+        logger.warning(f"{self.name} is no longer being processed.")
+        additional_message = self.metadata.get("additional_message", None)
```

Comment:
I see you always explicitly specify `None` as `default` arg value.
You know you don't have to do that, don't you?

Reply:
you are right, i confused it with getattr()

fixed

Reply:
there are more places where you set it